### DISCUSSION TITLE
MultiSelect renderValue prop

### DISCUSSION THREAD

**Discussion by jacobwilson41** [mantinedev/mantine#6051] [Feature requests]
Exactly like the `renderOption` prop but for when the option is **_selected_**

| Name | Type | Description|
|-------|------|------------|
`renderOption`| `((item: ComboboxLikeRenderOptionInput<ComboboxItem>) => ReactNode)`|A function to render content of the option, replaces the default content of the option|
|`renderValue` (new)| `((item: ComboboxLikeRenderOptionInput<ComboboxItem>) => ReactNode)`|A function to render content of the **_selected_** option, replaces the default content of the selected option|

**Comment by jacobwilson41** [comment]
I'm using the `renderOption` prop to custom render my options.  I want to be able to use the same rendering logic for selected items.

Unless I'm missing something, I can't find a way to do this.  Also the selected items displays the "value" of the option and I want it to display the "label"

https://github.com/mantinedev/mantine/assets/55633544/ab35985a-3f08-4313-9c9e-f8bc0e2d69aa

**Reply by rtivital** [reply]
You can implement that with Combobox component. It is not planned to be added to MultiSelect.

**Reply by jacobwilson41** [reply]
I just noticed that `MultiSelect` has an `inputContainer` prop.  What does that allow me to customize?

**Reply by rtivital** [reply]
https://mantine.dev/core/input/#inputcontainer

**Reply by antonfrolovsky** [reply]
In general, such a request sounds reasonable. I would also like to have a simpler way to display a value with custom markup.

```jsx
<Select 
  withAsterisk
  searchable
  label="Application"
  data={options}
  renderOption={renderOptions}
  renderValue={({option}) => (
    // custom markup here
  )}
/>
```

Creating a custom component with combobox is a rather overhead approach, since the basic functionality remains unchanged, everything except "custom" value.

@rtivital For what reasons are you rejecting this implementation, maybe I'm missing something?

**Comment by jjj201200** [comment]
I create one with cursor: [https://gist.github.com/jjj201200/15cdfc0fd5b8729d761cb79356a50b33](https://gist.github.com/jjj201200/15cdfc0fd5b8729d761cb79356a50b33)
